### PR TITLE
[master] Release core staging

### PR DESCRIPTION
**Pull request by Steve-White-UK**
## Description
Latest release from core_staging.
Metrics,
Tracing
P2P extension for tracing.
AccountStore reorg without Templates and optional sorting of serialisation,

## Backward Compatibility

- [X] This is not a breaking change
- [] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [X] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -19,17 +19,17 @@ spec:
     tty: true
     resources:
       requests:
-        ephemeral-storage: "20Gi"
+        ephemeral-storage: "30Gi"
```
Comment (n-hutton) [inline Jenkinsfile:22]: Is this neccesary as part of the PR?

File: src/libData/AccountStore/AccountStoreSCEvm.cpp [hunk 2]
```diff
@@ -34,20 +43,73 @@
...
 #include "libUtils/TxnExtras.h"
 
-template <class MAP>
-void AccountStoreSC<MAP>::EvmCallRunner(const INVOKE_TYPE /*invoke_type*/,  //
+namespace {
+
+// A temporary class
+
+template <typename T>
+class ArtemTextMapCarrier
```
Comment (n-hutton) [inline src/libData/AccountStore/AccountStoreSCEvm.cpp:54]: Not usual to have someone's name as part of a class

File: src/libData/AccountStore/AccountStoreSCEvm.cpp [hunk 3]
```diff
@@ -64,30 +126,73 @@ void AccountStoreSC<MAP>::EvmCallRunner(const INVOKE_TYPE /*invoke_type*/,  //
...
+      {  // example for Artem - retreving the context for shipment over the
+         // wire.
+        auto current_ctx = context::RuntimeContext::GetCurrent();
+        ArtemTextMapCarrier<std::map<std::string,std::string>> carrier;
+        auto prop = context::propagation::GlobalTextMapPropagator::
+            GetGlobalPropagator();
+        prop->Inject(carrier, current_ctx);
+
+        for (auto ting : carrier.data_) {
+          std::cout << "key :" << ting.first << " ";
```
Comment (n-hutton) [inline src/libData/AccountStore/AccountStoreSCEvm.cpp:147]: is this production code? Looks like debug

File: src/libData/AccountStore/AccountStoreTemp.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  AccountStore& m_parent;
+
+  friend class AccountStore;
+
+ public:
+  AccountStoreTemp(AccountStore& parent);
+
+  bool DeserializeDelta(const zbytes& src, unsigned int offset);
+
+  /// Returns the Account associated with the specified address.
```
Comment (n-hutton) [inline src/libData/AccountStore/AccountStoreTemp.h:36]:
```suggestion
  // Returns the Account associated with the specified address.
```

File: src/libData/AccountStore/services/evm/EvmProcessContext.h [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+#include "libData/AccountData/TransactionReceipt.h"
+#include "libUtils/Evm.pb.h"
+#include "libUtils/EvmUtils.h"
+
+struct EvmProcessContext {
+  const uint64_t& GetBlockNumber();
+
+ public:
+  /*
+   *   EvmProcessContext(const uint64_t& blkNum, const Transaction& txn,
```
Comment (n-hutton) [inline src/libData/AccountStore/services/evm/EvmProcessContext.h:58]: delete?

File: src/libData/AccountStore/services/evm/EvmProcessContext.h [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+#ifndef ZILLIQA_SRC_LIBDATA_ACCOUNTSTORE_SERVICES_EVM_EVMPROCESSCONTEXT_H_
+#define ZILLIQA_SRC_LIBDATA_ACCOUNTSTORE_SERVICES_EVM_EVMPROCESSCONTEXT_H_
+
+#include <memory>
+
+class Transaction;
+struct TxnExtras;
+
+/* EvmProcessContext
```
Comment (n-hutton) [inline src/libData/AccountStore/services/evm/EvmProcessContext.h:26]: note: duplicated in cpp file

File: src/libDirectoryService/DirectoryService.cpp [hunk 7]
```diff
@@ -881,7 +882,12 @@ bool DirectoryService::UpdateDSGuardIdentity() {
     }
   }
 
-  P2PComm::GetInstance().SendMessage(peerInfo, updatedsguardidentitymessage);
+  // TODO use distributed traces from here?
+  bool inject_trace_context = false;
+
+  P2PComm::GetInstance().SendMessage(peerInfo, updatedsguardidentitymessage,
```
Comment (n-hutton) [inline src/libDirectoryService/DirectoryService.cpp:888]: can you explain what is happening here? Also `updatedsguardidentitymessage` has no camel case which seems odd

File: src/libLookup/Lookup.cpp [hunk 8]
```diff
@@ -697,7 +699,11 @@ void Lookup::SendMessageToLookupNodes(const zbytes& message) const {
     }
   }
 
-  P2PComm::GetInstance().SendBroadcastMessage(allLookupNodes, message);
+  // TODO use distributed traces from here?
```
Comment (n-hutton) [inline src/libLookup/Lookup.cpp:702]: this todo is copied across the PR many times

File: tests/Persistence/Test_TxPersistence.cpp [hunk 9]
```diff
@@ -20,6 +20,7 @@
 #include <thread>
 #include <vector>
 
+#include "depends/libDatabase/LevelDB.h"
```
Comment (n-hutton) [inline tests/Persistence/Test_TxPersistence.cpp:23]: Why does this need to be here? Seems unrelated

File: tests/Persistence/Test_ExtSeedPubKeys.cpp [hunk 10]
```diff
@@ -22,7 +22,7 @@
 
 #include "libPersistence/BlockStorage.h"
 #include "libTestUtils/TestUtils.h"
-//#include "libUtils/TimeUtils.h"
+// #include "libUtils/TimeUtils.h"
```
Comment (n-hutton) [inline tests/Persistence/Test_ExtSeedPubKeys.cpp:25]:
```suggestion
// #include "libUtils/TimeUtils.h"
```
```suggestion
```

**Review by n-hutton — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (Steve-White-UK) [reply Jenkinsfile:22]: It was necessary to build the open telemetry on Jenkins, otherwise, the build would fail as it ran out of diskspace, Francesco upped it, I increased it again the next time we ran out of space, this is not a runtime feature, merely just enough to get through ci.

File: src/libData/AccountStore/AccountStoreSCEvm.cpp (hunk 2 above)
Comment (Steve-White-UK) [reply src/libData/AccountStore/AccountStoreSCEvm.cpp:54]: That code should have been deleted, nice catch, it was merely an example of how exporting for a distributed trace is handled.

Will gladly remove.

File: src/libData/AccountStore/AccountStoreSCEvm.cpp (hunk 3 above)
Comment (Steve-White-UK) [reply src/libData/AccountStore/AccountStoreSCEvm.cpp:147]: will be removed

File: src/libLookup/Lookup.cpp (hunk 8 above)
Comment (Steve-White-UK) [reply src/libLookup/Lookup.cpp:702]: Talk pvt.

File: src/libDirectoryService/DirectoryService.cpp (hunk 7 above)
Comment (Steve-White-UK) [reply src/libDirectoryService/DirectoryService.cpp:888]: Talk pvt.

File: tests/Persistence/Test_TxPersistence.cpp (hunk 9 above)
Comment (Steve-White-UK) [reply tests/Persistence/Test_TxPersistence.cpp:23]: probably overlooked in merge not anything we wished to change.

Comment (Steve-White-UK) [reply tests/Persistence/Test_TxPersistence.cpp:23]: corrected, we have clion too :-)
